### DISCUSSION TITLE
[SUGGESTION] Other devs/projects to involve

### DISCUSSION THREAD

**Discussion by MarcoRavich** [Amir-BK/unDAW#23] [Ideas]
We're pretty glad to pushed some kind of exchange with @naliMusic, so here's some other - maybe - interesting devs/projects to "involve":
- @gtreshchev of course: https://github.com/gtreshchev
- @MikyMikeMusic's [Cpp-Codes-For-UnrealEngine](https://github.com/MikyMikeMusic/My-Cpp-Codes-For-UnrealEngine);
- @kwstasg's [WAC](https://github.com/kwstasg/WAC);
- @jalc96's [audio_recorder_unreal_engine](https://github.com/jalc96/audio_recorder_unreal_engine);
- @spicynun's [UE4GainPlugin](https://github.com/spicynun/UE4GainPlugin);

- @Lahe's [AudioVisual](https://github.com/Lahe/AudioVisual);
- @AndreaCatania's [Goddess Sound Visualizer](https://github.com/AndreaCatania/GoddessSoundVisualizer)
- @amstaddigital's [eXiSoundVis](https://github.com/amstaddigital/eXiSoundVis);

Hope that helps/inspires !

**Comment by Mike014** [comment]
Hi Amir-BK and team,

Thank you for considering me and my project `Cpp-Codes-For-UnrealEngine` as
a potential collaborator for the `unDAW` project. I'm interested in
learning more about how you plan to involve my work and how we can
potentially collaborate.

Please let me know the details of what you have in mind, and I'd be happy
to discuss further.

Best regards,
MikyMikeMusic

Il mar 18 giu 2024, 11:05 F.O.R.A.R.T. npo ***@***.***> ha
scritto:

**Comment by Amir-BK** [comment]
Hey @MikyMikeMusic nice to e-meet you :)

unDAW is really just something I've been doing on my own, hoping someone finds use in it, it's really not "a big deal", if it came off that way, the guys from the https://github.com/forart/HyMPS project curate open source music projects and noticed unDAW, suggested I look into some other repo and I wound up forking it and making some upgrades to it (which came off pretty cool in the end I think). 

So, really that's the background, there's nothing official! It's just a hobby project that some UE5 stuff in it, it has 0 users that I know of right now so it's really not a big deal! WITH THAT SAID, if you took a look at the repo and you're interested in and what like to contribute I'm super into that, creating a semi functional DAW within unreal 5 is certainly possible and can even have some cool features (as already demonstrated fully modular signal paths per channel etc) but it's a lot of work, there's definitely a lot to do.

More specifically! Looking at your repo, one thing I think is super interesting and could be pretty 'mindblowing' from an audio sense would be to 'spatialize' multiple sources inside a single metasound and attaching those sources to the transforms (and velocities) of actors in the scene, I see that you've already done a lot of work on doppler effects and the such which is probably a great starting point.  it's take some figuring out for sure. 

So, if any of this interests you, let me know, so far I've been the sole contributor to the project so it'd take some figuring out (and onboarding!)

Cheers,
Amir

**Comment by Mike014** [comment]
Hey Amir,

Nice to e-meet you too! Thanks for sharing the background on `unDAW`. I'm
really impressed by how you've developed it as a personal project with
innovative features. Count me in—I'm genuinely interested in exploring how
we could push the boundaries further.

Your idea of spatializing multiple sources within a single metasound and
tying them to actor transforms in the scene is fascinating. We could also
consider handling certain Unreal Engine classes in C++ that aren't
accessible through Blueprints, or even integrating third-party plugins
built with the JUCE framework. There's a lot of potential for something
very compelling.

Let's discuss how we can move forward with these ideas. I'm excited about
the possibility of collaborating on `unDAW` and contributing to its growth
and development.

Looking forward to brainstorming with you and being part of this exciting
journey!

Cheers,
MikyMikeMusic

Il mar 18 giu 2024, 20:28 Amir Ben-Kiki ***@***.***> ha
scritto:

**Comment by Mike014** [comment]
Hi Amir, how are you? I wanted to ask you if you finally considered
bringing in as a project collaborator for a DAW.
Thank you very much for your reply,
Kind regards

Il giorno mar 18 giu 2024 alle ore 22:40 Mik Grimaldi <
***@***.***> ha scritto:

**Reply by Amir-BK** [reply]
Hey Michele, I'm doing well, what did you have in mind? with an open source repo there are many ways to collaborate, obviously for a solo project I don't have onboarding materials or even any sort of roadmap, I think it's like, the way to collaborate is just to start, have you explored the project so far and found something that you think needs to be improved? It's usually just best to pick something small and make a pull request, establish some 'mutual workflows' on the codebase, imo. 

I think that the spatial audio aspects might be a bit challenging to integrate (as a 'first issue'), but it's really up to you, there is already a lot going on in unDAW and if we want to introduce some spatialization capabilities there are definitely some exploratory tests that can be done.

Right now I'm in the process of making support for "send busses" inside the 'session metasound' by introducing a mixer node that can accept an arbitrary amount of pins, this will create the infrastructure for "daw like send busses" which are cool for stereo mixing but are not really super coherent in spatialization context, I think that giving thought to the "spatialization workflow" given the stuff that already exists is important, ideally I think we'd like to have capabilities such as "have both diagetic and non-diagetic sources in the same session" and "smooth fade from spatialize to non-spatial mixing", and of course need to think about how to attach to the in game transforms, though that one is probably the easiest to achieve with a dedicated actor component.

There are many places to start! There's much that can be improved, if you have some specific class/functionality you'd like us to review together I'm very open to that.

**Comment by Mike014** [comment]
Absolutely, just let me understand how you're working, did you write
Metasound in C++? I was taking a look at the project and I noticed that you
divided them into modules, which one do you think could be more right to be
able to start working on it? If you have time, give me an idea of how you
structured the work.

Il giorno sab 29 giu 2024 alle ore 12:23 Amir Ben-Kiki <
***@***.***> ha scritto:
